### DISCUSSION TITLE
budibaseDB quota limits?

### DISCUSSION THREAD

**Discussion by thomaskamalakis** [Budibase/budibase#6289] [Help] [answered]
Hi all

  I maybe hitting some budibaseDB quota limits. We have a rather large table "courses" containing descriptions of the courses offered by our Department containing titles in greek/english, descriptions, examination details, etc. When I create a simple provider for displaying the rows of the table and then a paginated table element inside the provider using the existing screen templates, thing go fine when I navigate in the first 49 rows but then when I hit row 50 the rows disappear from the preview.

  Here is the log of the couchdb service when this happens:

```
couchdb-service_1     | [info] 2022-06-13T10:27:45.435710Z couchdb@127.0.0.1 <0.961.5> -------- OS Process #Port<0.9477> Log :: function raised exception (new TypeError("doc.email is undefined", "<unknown>", 3)) with doc._id usage_quota
```

  Has anyone seen this?

  BR

  Thomas

**Comment by melohagan** [comment]
Hi @thomaskamalakis 

Whilst they should, could you double check that all of the users in the users list have an email address please.
I.e.
![Screenshot 2022-06-13 at 14 33 03](https://user-images.githubusercontent.com/101575380/173365277-25470bca-721d-4547-bc21-8a52dc3551f0.png)

And also check the users table for good measure.
I.e.
![Screenshot 2022-06-13 at 14 34 26](https://user-images.githubusercontent.com/101575380/173365527-4ada5dc5-c85a-4f47-a872-bd480c46bee7.png)

We don't think it's an issue with quotas.

**Reply by thomaskamalakis** [reply]
Dear @melohagan 

  Thanks, for some reason there was a user with email equal to "email" which I deleted. The error persists though.

**Comment by thomaskamalakis** [comment]
@melohagan, Interestingly enough, this also happens when I import the app in a fresh budibase setup.

Exported App is [here](https://drive.google.com/file/d/1jeq50Bw1fXmGQT2LPLvjRw5TbpII7uGR/view?usp=sharing), issue is in the Power Screen /

If I change the New Data Provider Limit to 49, rows magically appear :-)

![image](https://user-images.githubusercontent.com/25413968/173389104-ca63258a-7fcf-4a63-af36-75c5d4a19961.png)

![image](https://user-images.githubusercontent.com/25413968/173389223-ad41d773-e1f3-45ae-bd3e-9deca49412e1.png)

**Comment by thomaskamalakis** [comment, marked as answer]
OK so this may be an issue with the data after all.

I cleared up the course data using

```
re.sub("[^a-zA-Z0-9α-ωά-ώΑ-Ω +-/*.,\\n]","", field_value)
```
and then re-imported them in budibaseDB using the API. It does not seem to have a problem now - will conduct some more tests this week. Maybe some character sequence was causing an issue

Thanx!

**Reply by melohagan** [reply]
Brilliant, thanks for sharing. If you do find the cause please let us know!
